user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1216,15 +1240,97 @@ define([
         obj: $target,
         time: (new Date()).getTime()
       });
+      $(".folder.selected").children(".element").attr("tabindex",-1).attr("aria-selected",false);
       $(".folder.selected").removeClass("selected");
       $(".folder.secondarySelected").removeClass("secondarySelected");
       $target.addClass("selected");
+      $target.children(".element").attr("tabindex",0).attr("aria-selected",true);
       //deselect any files
       $(".file.selected").removeClass("selected");
       depth = $target.attr("path").split("/").length;
       event.stopPropagation();
     },
 
+    keyDownFolder: function (event) {
+      let keyCode = event.which || event.keyCode;
+      if (keyCode === 13 || keyCode === 32){
+        // ENTER , SPACE
+        this.clickFolder(event);
+      }else if (keyCode === 40){
+        // DOWN Arrow
+        let nextElement = this.getNextElementToFocus($(event.currentTarget));
+        if(nextElement !== undefined){
```

Comment:
See comments below, and test for (!== `null`).

Reply:
done